server/src/services/plugin-execution.service.ts

```diff
@@ -0,0 +1,214 @@
+  private loadedPlugins: Map<string, ExtismPlugin> = new Map();
+  private hostFunctions!: PluginHostFunctions;
+
+  @OnEvent({ name: 'AppBootstrap' })
+  async onBootstrap() {
+    this.hostFunctions = new PluginHostFunctions(
+      this.assetRepository,
+      this.albumRepository,
+      this.accessRepository,
+      this.cryptoRepository,
+      this.logger,
+    );
+    await this.loadCorePlugins();
+  }
+
+  private async loadCorePlugins() {
+    const plugins = await this.pluginRepository.getAllPlugins();
+    for (const plugin of plugins) {
+      try {
+        const pluginPath = resolve(process.cwd(), '..', plugin.wasmPath);
+        this.logger.debug(`Loading plugin: ${plugin.name} from ${pluginPath}`);
+
+        const extismPlugin = await newPlugin(pluginPath, {
+          useWasi: true,
+          functions: this.hostFunctions.getHostFunctions(),
+        });
+
+        this.loadedPlugins.set(plugin.id, extismPlugin);
+        this.logger.log(`Successfully loaded plugin: ${plugin.name}`);
+      } catch (error) {
+        this.logger.error(`Failed to load plugin ${plugin.name}:`, error);
+      }
+    }
+  }
+
+  @OnEvent({ name: 'AssetCreate' })
+  async handleAssetCreate({ asset }: ArgOf<'AssetCreate'>) {
+    const trigger = pluginTriggers.find((trigger) => trigger.type === PluginTriggerType.AssetCreate);
+    if (!trigger) {
+      return;
+    }
+
+    const workflows = await this.workflowRepository.getWorkflowsByTrigger(PluginTriggerType.AssetCreate);
+    if (workflows.length === 0) {
+      return;
+    }
+
+    const jobs: JobItem[] = workflows.map((workflow) => ({
+      name: JobName.WorkflowRun,
+      data: {
+        id: workflow.id,
+        type: PluginTriggerType.AssetCreate,
+        event: {
+          assetId: asset.id,
+        },
+      } as IWorkflowJob<PluginTriggerType.AssetCreate>,
+    }));
+
+    await this.jobRepository.queueAll(jobs);
+    this.logger.debug(`Queued ${jobs.length} workflow execution jobs for asset ${asset.id}`);
+  }
+
+  @OnJob({ name: JobName.WorkflowRun, queue: QueueName.Workflow })
+  async handleWorkflowRun({ id: workflowId, type, event }: JobOf<JobName.WorkflowRun>): Promise<JobStatus> {
+    try {
+      const workflow = await this.workflowRepository.getWorkflow(workflowId);
+      if (!workflow) {
+        this.logger.error(`Workflow ${workflowId} not found`);
+        return JobStatus.Failed;
+      }
+
+      const workflowFilters = await this.workflowRepository.getFilters(workflowId);
+      const workflowActions = await this.workflowRepository.getActions(workflowId);
+
+      switch (type) {
+        case PluginTriggerType.AssetCreate: {
+          const data = event as WorkflowData[PluginTriggerType.AssetCreate];
+          const asset = await this.assetRepository.getById(data.assetId);
+          if (!asset) {
+            this.logger.error(`Asset ${data.assetId} not found`);
+            return JobStatus.Failed;
+          }
```

Comment:
I think we should maybe just send the asset data in the event and avoid database lookups here. If the event is supposed to run after metadata extraction has put correct information in the asset table, then it can use a different event.